### ISSUE TITLE
`arr[arr.nonzero()]` does not work

### ISSUE THREAD

**Issue by eric-wieser** [pydata/sparse#295]
```
In [7]: m
Out[7]: <COO: shape=(32, 32, 32), dtype=float64, nnz=1024, fill_value=0.0>

In [6]: m[m.nonzero()]
IndexError: Only indices with at most one iterable index are supported.
```

**Comment by hameerabbasi** [comment]
Advanced indexing with more than one iterable index is not currently supported. `m.nonzero()` returns a tuple of three iterable indices... Therefore, the error message is valid. For `COO`, you can use `arr.data` instead.

We are waiting on NEP-21 to pass before we implement generalized advanced indexing.

**Comment by hameerabbasi** [comment]
Duplicate of #114.
